integration-cli/docker_cli_pause_test.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestPause(t *testing.T) {
+	defer deleteAllContainers()
+
+	name := "testeventpause"
+	out, _, _ := dockerCmd(t, "images", "-q")
+	image := strings.Split(out, "\n")[0]
+	dockerCmd(t, "run", "-d", "--name", name, image, "sleep", "2")
+
+	dockerCmd(t, "pause", name)
+	pausedContainers, err := getSliceOfPausedContainers()
+	if err != nil {
+		t.Fatalf("error thrown while checking if containers were paused: %v", err)
+	}
+	if len(pausedContainers) != 1 {
+		t.Fatalf("there should be one paused container and not", len(pausedContainers))
+	}
+
+	dockerCmd(t, "unpause", name)
+
+	eventsCmd := exec.Command(dockerBinary, "events", "--since=0", fmt.Sprintf("--until=%d", time.Now().Unix()))
+	out, _, _ = runCommandWithOutput(eventsCmd)
+	events := strings.Split(out, "\n")
+	if len(events) <= 1 {
+		t.Fatalf("Missing expected event")
+	}
+
+	pauseEvent := strings.Fields(events[len(events)-3])
+	unpauseEvent := strings.Fields(events[len(events)-2])
+
+	if pauseEvent[len(pauseEvent)-1] != "pause" {
+		t.Fatalf("event should be pause, not %#v", pauseEvent)
+	}
+	if unpauseEvent[len(unpauseEvent)-1] != "unpause" {
+		t.Fatalf("event should be unpause, not %#v", unpauseEvent)
+	}
+
+	waitCmd := exec.Command(dockerBinary, "wait", name)
+	if waitOut, _, err := runCommandWithOutput(waitCmd); err != nil {
+		t.Fatalf("error thrown while waiting for container: %s, %v", waitOut, err)
+	}
+
+	logDone("pause - pause/unpause is logged")
+}
+
+func TestPauseMultipleContainers(t *testing.T) {
+	defer deleteAllContainers()
```

Comment:
well so paused containers we cant send a sig to, its why we didnt have
pause tests for so long

On Fri, Feb 6, 2015 at 3:55 PM, Doug Davis <notifications@github.com> wrote:

> In integration-cli/docker_cli_pause_test.go
> <https://github.com/docker/docker/pull/10468#discussion_r24281597>:
>
> > +		t.Fatalf("event should be pause, not %#v", pauseEvent)
> > +	}
> > +	if unpauseEvent[len(unpauseEvent)-1] != "unpause" {
> > +		t.Fatalf("event should be unpause, not %#v", unpauseEvent)
> > +	}
> > +
> > +	waitCmd := exec.Command(dockerBinary, "wait", name)
> > +	if waitOut, _, err := runCommandWithOutput(waitCmd); err != nil {
> > +		t.Fatalf("error thrown while waiting for container: %s, %v", waitOut, err)
> > +	}
> > +
> > +	logDone("pause - pause/unpause is logged")
> > +}
> > +
> > +func TestPauseMultipleContainers(t *testing.T) {
> > +	defer deleteAllContainers()
>
> wow - deleteAllContainers() won't kill paused ones? I didn't know that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10468/files#r24281597>.
>